### PR TITLE
feat(builtins): add vacuum

### PR DESCRIPTION
**Pull request by cbochs**
This PR adds support for diagnostics using Vacuum, a fast OpenAPI linter written in Go.

### CODE REVIEW

File: lua/null-ls/builtins/diagnostics/vacuum.lua [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    generator_opts = {
+        command = "vacuum",
+        args = {
+            "spectral-report",
+            "--stdin",
+            "--stdout",
+        },
+        format = "json",
+        to_stdin = true,
+        ignore_stderr = true,
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/vacuum.lua:24]: Quick question - is this necessary? What happens to diagnostics if the linter is unable to parse the source file, for example?

Comment (cbochs) [reply lua/null-ls/builtins/diagnostics/vacuum.lua:24]: Good question. I'm not entirely certain. In fact, this was mostly copied from [lua/null-ls/builtins/diagnostics/spectral.lua](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/lua/null-ls/builtins/diagnostics/spectral.lua), which is a very similar tool. I can look into this if you would like

Comment (jose-elias-alvarez) [reply lua/null-ls/builtins/diagnostics/vacuum.lua:24]: It's probably not worth stressing about, if it causes issues we can look into it later.

**Comment by jose-elias-alvarez** [comment]
Thanks!

**Pull request merged**